odoo/addons/test_new_api/tests/test_properties.py

```diff
@@ -0,0 +1,1064 @@
+# -*- coding: utf-8 -*-
+import json
+
+from unittest.mock import patch
+
+from odoo import Command
+from odoo.exceptions import AccessError, CacheMiss
+from odoo.tests.common import Form, TransactionCase
+from odoo.tools import mute_logger
+
+
+class PropertiesCase(TransactionCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+
+        cls.user = cls.env['res.users'].search([], limit=1)
+        cls.partner = cls.env['res.partner'].search([], limit=1)
+
+        message_properties_1 = [
+            {
+                'id': 'name',
+                'string': 'Name',
+                'type': 'char',
+                'default': 'Default Name',
+            }, {
+                'id': 'partner_id',
+                'string': 'Partner',
+                'type': 'many2one',
+                'model': 'res.partner',
+            },
+        ]
+        message_properties_2 = [
+            {
+                "id": "state",
+                "type": "selection",
+                "string": "Status",
+                "selection": [("draft", "Draft"), ("progress", "In Progress"), ("done", "Done")],
+                "default": "draft",
+            }
+        ]
+
+        cls.discussion_1 = cls.env['test_new_api.discussion'].create({
+            'name': 'Test Discussion',
+            'message_properties': message_properties_1,
+            'participants': [Command.link(cls.user.id)],
+        })
+        cls.discussion_2 = cls.env['test_new_api.discussion'].create({
+            'name': 'Test Discussion',
+            'message_properties': message_properties_2,
+            'participants': [Command.link(cls.user.id)],
+        })
+
+        cls.message_1 = cls.env['test_new_api.message'].create({
+            'name': 'Test Message',
+            'discussion': cls.discussion_1.id,
+            'author': cls.user.id,
+            'custom_properties': {'name': 'Test', 'partner_id': [cls.partner.id, 'res.partner']},
+        })
+
+        cls.env.flush_all()
+
+        cls.message_2 = cls.env['test_new_api.message'].create({
+            'name': 'Test Message',
+            'discussion': cls.discussion_1.id,
+            'author': cls.user.id,
+        })
+        cls.message_3 = cls.env['test_new_api.message'].create({
+            'name': 'Test Message',
+            'discussion': cls.discussion_2.id,
+            'author': cls.user.id,
+        })
+
+    def test_json_field(self):
+        author = self.env['res.users'].search([], limit=1)
+
+        json_values = [
```

Comment:
If you move the test into another field, feel free to use completely different values, that do not look like field definitions, that way we don't confuse the fields.
(You can even go wild and use arrays within the values to make it look complete).